File: Python/challenge_4.py
#Challenge 4 (use python 2) ini adalah membuat output seperti berikut:
# 1 2 3 4 5 6
# 1 2 3 4 5
# 1 2 3 4
# 1 2 3
# 1 2
# 1
# 1 2
# 1 2 3
# 1 2 3 4
# 1 2 3 4 5
# 1 2 3 4 5 6

lebar = input("Lebar : ")
limit = lebar
awal = False

while (True):
	for i in range(lebar):
		print(i+1),	

	print("")
	if(not awal):
		lebar -= 1
	if(lebar == 0):
		awal = True
	if(awal):
		if lebar == 0:
			lebar += 2
		else:
			lebar += 1
	if(lebar > limit):
		print("Selesai")
		break

File: javascript/challenge_4.js
const readlineSync = require('readline-sync')

let lebar = readlineSync.questionInt("Lebar : ");
let limit = lebar;
let awal = false;

while (true){
    let temp = ""; 
	for(let i = 1; i <= lebar; i++){
		temp += i+" ";
    }
    console.log(temp);	

	if(!awal){
		lebar--;
    }

	if(lebar === 0){
		awal = true;
    }

	if(awal){
		if(lebar === 0){
			lebar += 2;
        }else{
			lebar++;
        }
    }

	if(lebar > limit){
		console.log("Selesai");
		break
    }
}